### DISCUSSION TITLE
Why does "PassthroughCluster" appear in Kiali when using Kafka?

### DISCUSSION THREAD

**Discussion by nikk0** [kiali/kiali#5331] [General]
hi.
I started to use kiali 2 days ago to visualize my K8s where kafka is used. I noticed that a Passthrough cluster appears in kiali UI. This is confusing to me.
Why does this appear ?
Is there a way to get rid of ?

kiali 1.38
istio 1.13.2

Thanks in advance

**Comment by jmazzitelli** [comment]
(converting to a discussion, not a bug report)

See the FAQ on this question "[Why do I see traffic to PassthroughCluster?](https://kiali.io/docs/faq/graph/#why-do-i-see-traffic-to-passthroughcluster)"

**Comment by jmazzitelli** [comment]
Side comment: Here's details on work that was done a couple years ago on getting Kafka use cases better integrated into Kiali - might be a good read for some additional background on kafka + kiali: https://github.com/kiali/kiali/issues/2197 .... and here is an article Joel wrote about Kiali visualizing Kafka traffic - another good read for additional info: https://medium.com/kialiproject/trace-my-mesh-part-3-3-e1e64795e61f

**Comment by nikk0** [comment]
Thanks for your feedback! 
As far as I can understand, there no out-of-the-box support for "crystal clear" kiali visualization of the traffic between kafka and its producers/consumers.

**Reply by jmazzitelli** [reply]
Any ideas for "crystal clear" visualization enhancements that can be done to Kiali can be submitted to the Kiali team in a [github enhancement request](https://github.com/kiali/kiali/issues/new?labels=enhancement). The Kiali team doesn't use Kafka regularly so we would require community assistance (and ideally actual code contributions) to move that kind of feature set forward.

**Comment by jmazzitelli** [comment]
Note also the graph you see (particularly the traffic flowing into "PassthroughCluster") is simply how Istio categorizes Kafka traffic in the Istio telemetry. So, in all likelihood, any enhancements to better visualize this in Kiali will also require some out-of-box support/enhancements to Istio as well. Kiali mainly graphs the Istio telemetry as-is - what the Istio telemetry contains is what Kiali shows. Without Kafka-specific metadata collected by Istio and stored in its telemetry, it may be difficult to get Kiali to do what you want.

**Reply by jmazzitelli** [reply]
re: the requirement that Istio provide some out of box support to allow Kiali to better support Kafka visualization -- see this Istio github enhancement request : https://github.com/istio/istio/issues/30267

And note the +1 and 19 likes for the asking of Kafka support in Istio here: https://github.com/istio/istio/issues/7664#issuecomment-429585606)

So it has been asked in Istio - but it doesn't look like there is progress. That may also require community assistance and contribution to get it into Istio.
